### PR TITLE
Update scheduling report with filters

### LINKED ISSUE #52: Updates to Scheduling (Leaves) Report
- [x] add start semester / end semester dropdowns, ideally going back 5 years as an option
- [x] remove section number
- [x] add enrollment info in parens (35/100) after course. May eventually make that a hover state or something.
- [x] Add filter by course level - grad/undergrad. Undergrad are courses <5000, grad are >=5000
- [x] Add filtering by appointment type - faculty, teaching specialist, etc. This will be filtering on jobcode by employee, which I definitely need to add to the API
- [x] start visible part of the table centered on current semester (if not too complex)

### PR DESCRIPTION
**Pull request by jxjj**
This updates the Scheduling Report:
- adds start/end semester dropdowns
- removes section number from listed course
- adds enrollment info next to course
- adds filters for course level, faculty appointment type, and course type (e.g. IND courses)
- highlights the current semester in light gold, and scrolls to it after report run completes.

Closes #52

### CODE REVIEW

File: resources/js/api.ts [hunk 1]
```diff
@@ -77,15 +77,21 @@ export async function getGroupCoursesByTerm({
   groupId: number;
   termId: number;
 }) {
-  // const cacheKey: GetGroupByTermCacheKey = `${groupId}-${termId}`;
-  // if (getGroupByTermCache.has(cacheKey)) {
-  //   return getGroupByTermCache.get(cacheKey)!;
-  // }
+  // cache if we're in dev mode (so we don't have to wait for the API)
```
Comment (cmcfadden) [inline resources/js/api.ts:80]: Can you add a note about this to the readme? I can see getting tripped up by it at some point in the future.

**Comment by jxjj** [comment]
Sorry, I started drafting this PR, but didn't mean it for review yet. Wanted to talk through direction and functionality with you.

File: resources/js/api.ts (hunk 1 above)
Comment (jxjj) [reply resources/js/api.ts:80]: I could see that. I doubt I'll think to look at the readme. I'm thinking maybe we should drop it? It's served it's purpose for now. Or, if we keep it, maybe put it behind a better named env variable like `CACHE_API_RESPONSES_CLIENTSIDE`?  What do you think?

Comment (cmcfadden) [reply resources/js/api.ts:80]: I'm fine dropping it

**Pull request merged**